pkg/internal/codegen/parse/crd.go

```diff
@@ -65,14 +65,24 @@ func (b *APIs) parseCRDs() {
 							Group:   fmt.Sprintf("%s.%s", resource.Group, resource.Domain),
 							Version: resource.Version,
 							Names: v1beta1.CustomResourceDefinitionNames{
-								Kind:   resource.Kind,
-								Plural: resource.Resource,
+								Kind:       resource.Kind,
+								Plural:     resource.Resource,
+								ShortNames: []string{resource.ShortName},
 							},
 							Validation: &v1beta1.CustomResourceValidation{
 								OpenAPIV3Schema: &resource.JSONSchemaProps,
 							},
 						},
 					}
+
+					result, err := parseResourceAnnotation(resource.Type)
+					if err != nil {
+						log.Fatalf("failed to parse resource annotations, error: %v", err.Error())
+					}
+					resource.CRD.ObjectMeta.Name = result.Resource + "." + resource.Group + "." + resource.Domain
```

Comment:
it should be possible to *just* specify the shortname here and use the default for the resource name.

Reply:
I agree, we should support specifying *just* the shortname here because majority of the time users would like do that. So, we should derive the name/plural/shortname only if they were specified in the annotation.